src/runmode-af-packet.c

```diff
@@ -287,8 +287,16 @@ static void *ParseAFPConfig(const char *iface)
     }
 
     if (ConfGetChildValueWithDefault(if_root, if_default, "copy-iface", &out_iface) == 1) {
-        if (strlen(out_iface) > 0) {
-            aconf->out_iface = out_iface;
+        if (out_iface != NULL) {
+            if (strlen(out_iface) > 0) {
+                aconf->out_iface = out_iface;
+            }
+            if (strcmp(iface, out_iface) == 0) {
+                FatalError("Invalid config: interface (%s) and copy-iface (%s) can't be the same",
+                        iface, out_iface);
+            }
+        } else {
+            SCLogWarning("copy-iface corresponding to %s interface is %s", iface, out_iface);
```

Comment:
using `out_iface` here makes little sense as we know it will be null.

kind of wonder if we should have `ConfGetChildValueWithDefault` check for an empty value instead?

Reply:
> using out_iface here makes little sense as we know it will be null.

yeah. It came out as a nice `(null)` in the message so I let it be.

> kind of wonder if we should have ConfGetChildValueWithDefault check for an empty value instead?

I did that in https://github.com/OISF/suricata/pull/9335, QA does not seem to have any issues, so, creating a new PR with this approach in mind.